src/Chart.ts

```diff
@@ -0,0 +1,149 @@
+    const { height, width, margin } = bounds;
+ 
+    const svg = d3.create("svg")
+      .attr("id", getHtmlId(LayerType.Svg))
+      .attr("width", "100%")
+      .attr("height", "100%")
+      .attr("viewBox", `0 0 ${width} ${height}`)
+      .attr("preserveAspectRatio", "xMinYMin")
+      .attr("vector-effect", "non-scaling-stroke") as any as D3Selection<SVGSVGElement>;
+
+    const clipPath = svg.append("defs")
+      .append("svg:clipPath")
+      .attr("id", getHtmlId(LayerType.ClipPath)) as any as D3Selection<SVGClipPathElement>;
+    clipPath.append("svg:rect")
+      .attr("vector-effect", "non-scaling-stroke")
+      .attr("width", width - margin.right - margin.left)
+      .attr("height", height - margin.bottom - margin.top)
+      .attr("x", margin.left)
+      .attr("y", margin.top);
+
+    const baseLayer = svg.append('g')
+      .attr("id", getHtmlId(LayerType.BaseLayer))
+      .attr("vector-effect", "non-scaling-stroke")
+      .attr("clip-path", `url(#${getHtmlId(LayerType.ClipPath)})`);
+
+    const { x, y } = this.scales;
+    const scaleX = d3.scaleLinear()
+      .domain([x.start, x.end])
+      .range([ margin.left, width - margin.right ]);
+    const scaleY = d3.scaleLinear()
+      .domain([y.start, y.end])
+      .range([ height - margin.bottom, margin.top ]);
+    
+    const lineGen = d3.line<Point>()
+      .x(d => scaleX(d.x))
+      .y(d => scaleY(d.y));
+
+    const layerArgs: LayerArgs = {
+      id: this.id,
+      getHtmlId,
+      bounds,
+      globals: this.globals,
+      scaleConfig: {
+        linearScales: { x: scaleX, y: scaleY },
+        lineGen,
+        scaleExtents: this.scales
+      },
+      coreLayers: {
+        [LayerType.Svg]: svg,
+        [LayerType.ClipPath]: clipPath,
+        [LayerType.BaseLayer]: baseLayer
+      },
+      optionalLayers: []
+    };
+
+    baseElement.childNodes.forEach(n => n.remove());
+
+    this.optionalLayers.forEach(l => {
+      l.draw(layerArgs);
+      layerArgs.optionalLayers.push(l);
+    });
+
+    baseElement.append(layerArgs.coreLayers[LayerType.Svg].node()!);
+  };
+
+  appendTo = (baseElement: HTMLDivElement) => {
+    const defaultMargin = { top: 20, bottom: 20, left: 40, right: 40 };
+    const drawWithBounds = (width: number, height: number) => {
+      const bounds = { width, height, margin: defaultMargin };
+      this.draw(baseElement, bounds);
+    };
+
+    const { width, height } = baseElement.getBoundingClientRect();
+    drawWithBounds(width, height);
+
+    if (this.isResponsive) {
+      const resizeObserver = new ResizeObserver(entries => {
+        const { blockSize: height, inlineSize: width } = entries[0].borderBoxSize[0];
+        drawWithBounds(width, height);
+      });
+      resizeObserver.observe(baseElement);
+      window.addEventListener("resize", e => {
+        if (!e.view) return;
+        const { innerHeight: height, innerWidth: width } = e.view;
+        drawWithBounds(width, height);
+      });
```

Comment:
A bit of explanation would be nice here. Mostly because I can't remember why you need the window event listener separately from the resize observer!